homeassistant/components/music_assistant/__init__.py

```diff
@@ -174,3 +174,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         await mass_entry_data.mass.disconnect()
 
     return unload_ok
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    player_id = next(
+        (
+            identifier[1]
+            for identifier in device_entry.identifiers
+            if identifier[0] == DOMAIN
```

Comment:
Why don't we just check on startup if we have orphan devices? This way users never have to take any action in managing their devices

Reply:
Well, because there is possibility that a device is just unplugged/offline and then we throw away a device that may be added back later.

Reply:
But how do you do that on runtime then? If someone unplugs a device we delete it?

Reply:
Yeah, but then the user will do that manually, knowing what he's doing.

Reply:
No I mean

> If you remove a player(config) on the Music Assistant Server while the Home Assistsant integration is running, it will auto pickup that as an event and remove the device from HA as well

Reply:
Yeah, that is already in the code. We are just talking about the (edge) case here where the user removed a player while HA was offline.

Reply:
But isn't that the same case? Maybe someone unplugged it to move to another room or its offline, then we automatically remove the device from HA?

Reply:
No, that event is fired when they press the DELETE button in the MA interface.
Just like this PR adds to HA, a button to initiate the delete action which will be denied if the player is not orphan

Reply:
Strictly taken we can limit this PR to only remove the deleted players btw and see the extra delete feature as extra

Reply:
Right, so why don't we remove all the deleted devices by checking all the non-deleted devices and compare those with the ones in the device registry?

I strongly dislike this function, since 90% of the time, we can just automatically do it, and once that's there, there's no need for the function anymore

Reply:
But do we still need it if we check the deleted players at boot and listen to events? This would only not work if your set up is broken, but then you have more problems than just an orphan device

Reply:
Need is a strong word. It would be a feature to also be able to delete devices from the HA device page instead of only from the MA settings page. 

I'll adjust it and remove that logic and limit to only check the player configs at startup and compare against dev reg.

Reply:
Honestly, IMHO this one make sense 👍 

The best case: A user is able to delete a device that was missed. Worst case, the device returns later on.

Reply:
So leave it like this?

Reply:
I would say yes. But @joostlek seems to think differently. I'm just trying to add another opinion / point of view.

Reply:
So... I just added both.
At startup of the integration check for orphaned devices (no more player config present on the server) but also keep the manual delete function which calls the delete function on the server itself.

Reply:
Sounds good to me, let's see what @joostlek thinks 👍